unit2-apis-and-advanced-javascript/README.md

```diff
@@ -18,68 +18,68 @@
   - JSX and the benefits of writing HTML markup in javascript
   - Adding a reactjs task to your `gulpfile`
-  - Light introduction to ES6 (most of ReactJS + ReactJS documentation is written in es6)
-    - classes, extend keyword, arrow functions, spread operator
+  - Light introduction to ES6 (most of ReactJS + ReactJS documentation is written in es6) [_BENNO_: My concern is that there won't be enough time]
+    - classes, extend keyword, arrow functions, spread operator, promises
   - **Learning outcome**: Create simple To Do list using ReactJS
   - **Learning outcome**: Understand the difference between state and prop
-- Redux: 2 days
+- Redux: 1 day
   - Principles of redux: minimize state mutations, single source of truth, functional programming
   - Actions, Reducers, Stores, etc
   - **Learning outcome**: Build a Trello API based blog using ReactJS and Redux
- - D3.js: 1 day
-  - selections
-  - entering / exiting (ie: how to create/remove nodes as needed from incoming data)
-  - transitions
-  - **Learning outcome**: Given a d3.js component (we could build it for them), design and implement a react.js app that has many d3 graphs whose state is managed by Redux
+- Queues & Stacks: 1 day
+  - Queues: First in, First out. Waiting in line, scheduling systems
+  - Stacks: First in, Last out. Stack of plates, books, browser back button, function call stack
+  - How to build either using linked list. How to build a fixed-sized Queue using an array (c-style)
+  - **Learning outcome**: Understand which real-life situations map to which DS; Given a DS what are the operations it supports? Given some code, what is the state of the DS after some push/pop operations?
 
 ### Week 4
 
-### TODO: below is stuff to mix in
-
-5. Objects / HashTables
- - Syntax for JS objects
- - Common uses for key-value mappings
- - HashTable C-style implementation (array of linked lists) and the performance implications
- - **Learning outcome**: Write a JavaScript function that takes an input an array and returns a new array with only the unique elements from the input. This should be O(n) time and preserve the order of the input array
- - **Learning outcome**: Create a vehicle lookup implementation that lets users lookup vehicles by name and display the different features of the vehicle on the page.
-6. Queues & Stacks
- - Queues: First in, First out. Waiting in line, scheduling systems
- - Stacks: First in, Last out. Stack of plates, books, browser back button, function call stack
- - How to build either using linked list. How to build a fixed-sized Queue using an array (c-style)
- - **Learning outcome**: Understand which real-life situations map to which DS; Given a DS what are the operations it supports? Given some code, what is the state of the DS after some push/pop operations?
+- Major React Project using D3.js: 2 days
+  - Data fetched via AJAX from an API provided by us
```

Comment:
Yup sounds good. "Provided by us" doesn't mean "built custom". I've been using http://jsonplaceholder.typicode.com/ when writing challenges for graduates